### PR TITLE
Smart redirect

### LINKED ISSUE #1516: Create a smart redirect mechanism
### Describe the Issue

Looking at the data that we have about 404s, it's safe to say that need a smart redirect mechanism to reduce user's frustration when navigating through the website.

We already have a lot of redirects on the _redirects file, but a smart redirect would help alleviate this problem.

### PR DESCRIPTION
**Pull request by Deflaimun**
Resolves #1516 

First iteration of a smart redirect engine. 

The engine gets the ending URL and tries to smart redirect the user. 
If it finds the key in the sidebars.js file, sends the user to correct page.

All this info is sent to the NotFoundForm notifier.

As a protection for infinite loops, this info is saved on the sessionStorage. Keep in mind that the session Storage should get invalided after the end of session.

Still not working on :
- versioned docs

Preview: 
https://deploy-preview-1586--redpanda-documentation.netlify.app/

### CODE REVIEW

File: src/theme/NotFound.js [hunk 1]
```diff
@@ -3,18 +3,100 @@ import Translate, { translate } from "@docusaurus/Translate";
...
+let possibleNewRedirectRule;
+let possibleRedirect;
+
+function getDocIds(data) {
+  const docIds = [];
+
+  function iterateDocs(items) {
+    if (Array.isArray(items)) {
+      items.forEach((obj) => {
+        if (obj.type === "doc") {
```
Comment (shorgi) [inline src/theme/NotFound.js:20]: In sidebars.js, some docs aren't specified as `{type: doc, label: ..., id: ...}` but as the doc page itself (for example, `develop/http-proxy`). Do the items that are doc pages themselves have `obj.type === "doc"`?

**Review by shorgi — COMMENTED** [review]
Tested by going to https://deploy-preview-1586--redpanda-documentation.netlify.app/docs/get-started/quick-start, then changing its link to https://deploy-preview-1586--redpanda-documentation.netlify.app/docs/get-started/quick-start/kafka-clients to see if it'd auto-redirect to the kafka-clients page. It did do a smart redirect to the expected page, but only after displaying the 404 Not Found page.

**Comment by JakeSCahill** [comment]
Rather than redirecting, would it be a better UX to provide links to relevant pages on the 404 page?
This way, users still recognize that the URL they are using is not valid and we use the smart redirect logic to provide a list of similar pages that do exist.

File: src/theme/NotFound.js (hunk 1 above)
Comment (Deflaimun) [reply src/theme/NotFound.js:28]: Fixed! Good catch

**Review by shorgi — APPROVED** [review]
lgtm

**Pull request merged**